src/app/barter/components/user-transactions/user-transactions.component.ts

```diff
@@ -0,0 +1,185 @@
+  @ViewChild(MdbTablePaginationComponent, {static: true}) mdbTablePagination: MdbTablePaginationComponent;
+  @ViewChild(MdbTableDirective, {static: true}) mdbTable: MdbTableDirective;
+  elements: any = [];
+  previous: any = [];
+  headElements = ['Bidder', 'Owner', 'Offered Item', 'Owner Item', 'Action'];
+  transactionsElements = [];
+  modalRef: MDBModalRef;
+  offeredToMeCheck: boolean = false;
+  completedTransactionsCheck: boolean = false;
+  myOffersCheck: boolean = false;
+  currentUser: User;
+  maxTableRows: number = 5;
+
+  constructor(private cdRef: ChangeDetectorRef,
+              private activatedRoute: ActivatedRoute,
+              private modalService: MDBModalService,
+              private transactionStore$: Store<TransactionsState>,
+              private itemsStore$: Store<ItemsState>,
+              private userStore$: Store<UserState>,
+              private alertsService: AlertsService) {
+  }
+
+  ngOnInit() {
+    this.activatedRoute.data.subscribe((result) => {
+      result.data.forEach(transaction =>
+        this.elements.push({
+          id: transaction.id,
+          bidder: transaction.trader,
+          owner: transaction.owner,
+          offeredItem: transaction.traderItem,
+          ownerItem: transaction.ownerItem,
+          isCompleted: transaction.isCompleted,
+          inProgress: transaction.inProgress
+        }));
+
+      this.elements.sort((a, b) => a.isCompleted - b.isCompleted);
+
+      this.transactionsElements = this.elements;
+      this.setTableData();
+    });
+
+    this.userStore$.select(getUser).subscribe((user: User) => this.currentUser = user);
+  }
+
+  ngAfterViewInit() {
+    this.mdbTablePagination.setMaxVisibleItemsNumberTo(this.maxTableRows);
+    this.mdbTablePagination.calculateFirstItemIndex();
+    this.mdbTablePagination.calculateLastItemIndex();
+    this.cdRef.detectChanges();
+  }
+
+  showItemDetails(item: Item): void {
+    this.modalRef = this.modalService.show(ItemDetailsModalComponent, {
+      data: {
+        item,
+        isWatchMode: true
+      },
+      class: 'modal-lg'
+    });
+  }
+
+  acceptOffer(transaction): void {
+    this.itemsStore$.select(getTransactionItems(transaction.ownerItem.id, transaction.offeredItem.id)).pipe(first())
+      .subscribe((items: Item[]) => {
+        const firstItem: Item = {...items[0]};
+        const secondItem: Item = {...items[1]};
+
+        const firstItemOwnerId = firstItem.ownerId;
+        firstItem.ownerId = secondItem.ownerId;
+        secondItem.ownerId = firstItemOwnerId;
+
+        const transactionIndex = this.elements.indexOf(this.elements.find(element =>
+          (element.offeredItem.id === firstItem.id || element.offeredItem.id === secondItem.id) &&
+          ((element.ownerItem.id === firstItem.id || element.ownerItem.id === secondItem.id))));
+        this.elements[transactionIndex].isCompleted = true;
+
+        this.itemsStore$.dispatch(updateItem({item: firstItem}));
```

Comment:
After a trade, items should be removed from the home page. since it doesn't make sense to trade again - 

- [ ] use a boolean for completion as item status or something

Reply:
we need to think about it. right now there is a boolean for item called isLoaned.
We can use it, but I think we need to talk about it and maybe consider use it in item edit or in my items in order to display this item again. For now I leave it this way, and we will open a task in our next meeting